### ISSUE TITLE
Unable to Donate via PayPal

### ISSUE THREAD

**Issue by lunarius** [HabitRPG/habitica#2492]
Hello,

When using Chrome on a Mac, if I click on the PayPal button I'm taken to a blank page with "about:blank" in the URL bar.

When using Safari on a Mac, if I click on the PayPal button I'm taken to a blank page with "https://www.[object%20Object].paypal.com/cgi-bin/webscr" in the URL bar.

When using Firefox on a Mac, if I click on the PayPal button I'm taken nowhere; the click doesn't seem to register at all.

In all three browsers I am successfully given the option to pay via credit card on site.

Chrome is Version 32.0.1700.77
Safari is Version 7.0.1 (9537.73.11)
Firefox is Version 26.0
The Mac is Version 10.9.1

All extensions, mods, addons, etc., were disabled while I was testing this.

**Comment by colegleason** [comment]
Also seeing this on my (production) account. data-env is being set to an object somehow. Relevant line: https://github.com/HabitRPG/habitrpg/blob/003e21408020771e2a878060173596b050c0618d/views/shared/modals/buy-gems.jade#L33

**Comment by ghost** [comment]
Experiencing this same phenomenon on Chrome for Windows Vista.

**Comment by domi7777** [comment]
"about:blank" in url bar and empty page on Chrome win 7

**Comment by colegleason** [comment]
Sorry, I'll try to look into this today.

On Thu, Jan 23, 2014 at 8:04 AM, domi7777 notifications@github.com wrote:

> "about:blank" in url bar and empty page on Chrome win 7
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/HabitRPG/habitrpg/issues/2492#issuecomment-33125870
> .

## 

## Cole Gleason

Student, University of Illinois at Urbana-Champaign
Email: cg@colegleason.com
Website: colegleason.com

**Comment by colegleason** [comment]
Okay, for some reason on the like site, it seems that `env.NODE_ENV == 'production' ? '' : 'sandbox'` somehow results in an object. On my local instance in production mode I only get an empty string.

I do not understand.

**Comment by lunarius** [comment]
Is there a way that, in the meantime, users can donate via PayPal and still get gems?

**Comment by colegleason** [comment]
Unfortunately not as of right now. You can still Pay by Card (through Stripe, our credit card processor). I'm trying to figure this one out, sorry!

**Comment by colegleason** [comment]
@lefnire, what is the currently deployed commit for habitrpg and habitrpg shared? Maybe if I can replicate it exactly...

**Comment by scheidm** [comment]
I'm having the same issue on windows 8.1 firefox.

**Comment by lefnire** [comment]
Sorry guys, I'll jump in later today. On a trip with limited computer
access, more later

**Comment by lefnire** [comment]
Looks like [JavascriptButtons](https://github.com/paypal/JavaScriptButtons) has been updated. Far as I can tell, it's our use of `''` for production mode that's triggering this. I tried `www`, which is stated to be the "production mode default" instead of an empty string, but then it was linking to `https://www.www.paypal...` which was wrong. So for now, I'm just removing sandbox mode - I haven't used it to test locally for ages anyway.

@colegleason reason you couldn't replicate locally is the issue was in bower_components, do a `bower update`
